src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -909,7 +909,7 @@ RouterModifyTaskForShardInterval(Query *originalQuery,
 		 * Note that this is only the case with PG14 as the parameter doesn't exist
 		 * prior to that.
 		 */
-		shardRestrictionList = make_simple_restrictinfo(NULL,
+		shardRestrictionList = make_simple_restrictinfo(restriction->plannerInfo,
```

Comment:
we have a comment above saying that it is OK to pass NULL. We should probably change that.

Now, wondering, why haven't we passed the plannerInfo before? Do you have any clue? seems introduced here: https://github.com/citusdata/citus/commit/b01e7e884c9ab9bc8cf5520e76c6b2bf531fc8b2

Reply:
Let me investigate further

Reply:
> why haven't we passed the plannerInfo before?

I am fine with the explanation of the comment - apparently it wasn't necessary until PG16 since it was only used for placeholder vars.
However, in PG16, there are extra operations in `make_restrictinfo_internal` that are accessing PlannerInfo directly, hence crashing if we pass null.
```C
static RestrictInfo *
make_restrictinfo_internal(PlannerInfo *root, ....)
{
....
        /*
	 * Count the number of base rels appearing in clause_relids.  To do this,
	 * we just delete rels mentioned in root->outer_join_rels and count the
	 * survivors.  Because we are called during deconstruct_jointree which is
	 * the same tree walk that populates outer_join_rels, this is a little bit
	 * unsafe-looking; but it should be fine because the recursion in
	 * deconstruct_jointree should already have visited any outer join that
	 * could be mentioned in this clause.
	 */
	baserels = bms_difference(restrictinfo->clause_relids,
							  root->outer_join_rels);
	restrictinfo->num_base_rels = bms_num_members(baserels);
	bms_free(baserels);
....
}
```
Relevant PG16 commit https://github.com/postgres/postgres/commit/2489d76c4906f4461a364ca8ad7e0751ead8aa0d#diff-e045c41eda9686451a7993e91518e40056b3739365e39eb1b70ae438dc1f7c76R206